packages/next/build/webpack/plugins/build-manifest-plugin.ts

```diff
@@ -92,6 +153,32 @@ export default class BuildManifestPlugin {
         compilation.assets[BUILD_MANIFEST] = new RawSource(
           JSON.stringify(assetMap, null, 2)
         )
+
+        if (this.clientManifest) {
+          const clientManifestPath = `${CLIENT_STATIC_FILES_PATH}/${
+            this.buildId
+          }/_buildManifest.js`
+
+          compilation.assets[clientManifestPath] = new RawSource(
```

Comment:
This code needs to not run when not a modern build, right?

Reply:
This part is for the _buildManifest.js file, which needs to be created regardless of whether experimental.modern is true--it's needed for granular chunks, as well. The part down at line 170 is what we want to do only if modern build is enabled

Reply:
Yes but this plugin gets invoked twice AFAIK, so both branches would be hit. We just need to detect if it _is_ the modern build, not if the feature is enabled.

Reply:
I guess we can leave this as-is, it works. I don't want to be dogmatic.